presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -564,8 +564,164 @@ To find the `ROC curve <https://en.wikipedia.org/wiki/Receiver_operating_charact
+
+The following functions approximate the binary `differential entropy <https://en.wikipedia.org/wiki/Differential_entropy>`_.
+That is, for a random variable :math:`x`, they approximate
+
+.. math ::
+
+    H(x) = - \int x \log_2\left(f(x)\right) dx,
+
+where :math:`f(x)` is the partial density function of :math:`x`.
+
+.. function:: differential_entropy(sample_size, x)
+
+    Returns the approximate log-2 differential entropy from a random variable's sample outcomes. The function internally
+    creates a reservoir (see [Black2015]_), then calculates the
+    entropy from the sample results by approximating the derivative of the cumulative distribution
+    (see [Alizadeh2010]_).
+
+    ``sample_size`` (``long``) is the maximal number of reservoir samples.
+
+    ``x`` (``double``) is the samples.
+
+    For example, to find the differential entropy of ``x`` of ``data`` using 1000000 reservoir samples, use
+
+    .. code-block:: none
+
+         SELECT
+             differential_entropy(1000000, x)
+         FROM
+             data
+
+    .. note::
+
+        If :math:`x` has a known lower and upper bound,
+        prefer the versions taking ``(bucket_count, x, 1.0, "fixed_histogram_mle", min, max)``,
+        or ``(bucket_count, x, 1.0, "fixed_histogram_jacknife", min, max)``,
+        as they have better convergence.
+
+.. function:: differential_entropy(sample_size, x, weight)
+
+    Returns the approximate log-2 differential entropy from a random variable's sample outcomes. The function
+    internally creates a weighted reservoir (see [Efraimidis2006]_), then calculates the
+    entropy from the sample results by approximating the derivative of the cumulative distribution
+    (see [Alizadeh2010]_).
+
+    ``sample_size`` is the maximal number of reservoir samples.
+
+    ``x`` (``double``) is the samples.
+
+    ``weight`` (``double``) is a non-negative double value indicating the weight of the sample.
+
+    For example, to find the differential entropy of ``x`` with weights ``weight`` of ``data``
+    using 1000000 reservoir samples, use
+
+    .. code-block:: none
+
+         SELECT
+             differential_entropy(1000000, x, weight)
+         FROM
+             data
+
+    .. note::
+
+        If :math:`x` has a known lower and upper bound,
+        prefer the versions taking ``(bucket_count, x, weight, "fixed_histogram_mle", min, max)``,
+        or ``(bucket_count, x, weight, "fixed_histogram_jacknife", min, max)``,
+        as they have better convergence.
+
+.. function:: differential_entropy(bucket_count, x, weight, "fixed_histogram_mle", min, max) -> double
+
+    Returns the approximate log-2 differential entropy from a random variable's sample outcomes. The function
+    internally creates a conceptual histogram of the sample values, calculates the counts, and
+    then approximates the entropy using maximum-likelihood estimation (see [Beirlant2001]_).
+
+    ``bucket_count`` (``long``) is the number of histogram buckets.
```

Comment:
long is not a valid type in Presto; see https://prestodb.github.io/docs/current/language/types.html for the list of types